worker/uniter/metrics.go

```diff
@@ -45,11 +51,22 @@ func (t *timerChooser) getMetricsTimer(ch corecharm.Charm) TimedSignal {
 	return t.inactive
 }
 
+// getSendMetricsTimer returns the metrics timer we should be using, given the supplied
+// charm.
+func (t *timerChooser) getSendMetricsTimer(ch corecharm.Charm) TimedSignal {
+	metrics := ch.Metrics()
+	if metrics != nil && len(metrics.Metrics) > 0 {
+		return t.activeSender
+	}
+	return t.inactive
+}
+
 // NewMetricsTimerChooser returns a timerChooser for
 // collect-metrics.
 func NewMetricsTimerChooser() *timerChooser {
 	return &timerChooser{
-		active:   activeMetricsTimer,
-		inactive: inactiveMetricsTimer,
+		active:       activeMetricsTimer,
```

Comment:
For example, I think this ought to be `activeCollect: activeCollectMetricsTimer`.